Code/Editor/CryEditPy.cpp

```diff
@@ -147,15 +147,16 @@ namespace
                 return false;
             }
         }
+        const bool addToMostRecentFileList = false;
 
         auto previousDocument = GetIEditor()->GetDocument();
         QString previousPathName = (previousDocument != nullptr) ? previousDocument->GetLevelPathName() : "";
-        auto newDocument = CCryEditApp::instance()->OpenDocumentFile(levelPath.toUtf8().data());
+        auto newDocument = CCryEditApp::instance()->OpenDocumentFile(levelPath.toUtf8().data(), addToMostRecentFileList, COpenSameLevelOptions::ReopenLevelIfSame);
 
         // the underlying document pointer doesn't change, so we can't check that; use the path name's instead
 
         bool result = true;
-        if (newDocument == nullptr || newDocument->IsLevelLoadFailed() || (newDocument->GetLevelPathName() == previousPathName))
+        if (newDocument == nullptr || newDocument->IsLevelLoadFailed())
```

Comment:
The comment above this `the underlying document pointer doesn't change, so we can't check that; use the path name's instead` is not valid anymore, please remove.

Also, the result variable and if condition could go away and simplify it like this.

````
return 
    newDocument != nullptr &&
    !newDocument->IsLevelLoadFailed();
````

It reads better to what it's expecting to happen for the function to return true.